docs/usage/shoot_updates.md

```diff
@@ -0,0 +1,91 @@
+
+### In-Place vs. Rolling Updates
+
+If the Kubernetes patch version is changed then the upgrade happens in-place.
+This means that the shoot worker nodes remain untouched and only the `kubelet` process restarts with the new Kubernetes version binary.
+
+If the Kubernetes minor version is changed then the upgrade is done in a "rolling update" fashion, similar to how pods in Kubernetes are updated (when backed by a `Deployment`).
+The worker nodes will be terminated one after another and replaced by new machines.
+Existing workload will be gracefully drained and evicted from the old worker nodes to new worker nodes, respecting the configured `PodDisruptionBudget`s (see [Kubernetes documentation](https://kubernetes.io/docs/tasks/run-application/configure-pdb/)).
+
+#### Customize Rolling Update Behaviour of Shoot Worker Nodes
+
+The `.spec.provider.workers[]` list exposes two fields that you might configure based on your workload's needs: `maxSurge` and `maxUnavailable`.
+The same concepts [like in Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment) apply.
+Additionally, you might customize how the machine-controller-manager (abbrev.: MCM; the component instrumenting this rolling update) is behaving. You can configure the following fields in `.spec.provider.worker[].machineControllerManager`:
+
+* `machineDrainTimeout`: Timeout (in duration) used while draining of machine before deletion, beyond which MCM forcefully deletes machine (default: `10m`).
+* `machineHealthTimeout`: Timeout (in duration) used while re-joining (in case of temporary health issues) of machine before it is declared as failed (default: `10m`).
+* `machineCreationTimeout`: Timeout (in duration) used while joining (during creation) of machine before it is declared as failed (default: `10m`).
+* `maxEvictRetries`: Maximum number of times evicts would be attempted on a pod before it is forcibly deleted during draining of a machine (default: `10`).
+* `nodeConditions`: List of case-sensitive node-conditions which will change a machine to a `Failed` state after the `machineHealthTimeout` duration. It may further be replaced with a new machine if the machine is backed by a machine-set object (defaults: `KernelDeadlock`, `ReadonlyFilesystem` , `DiskPressure`).
+
+#### Other Rolling Update Triggers
+
+Some changes to your worker pool specification (`.spec.provider.workers[]`) will also result in a rolling update of your worker nodes even if you don't change the Kubernetes or machine image version.
+Those fields include:
+
+* `.spec.provider.workers[].machine.type`
+* `.spec.provider.workers[].volume.type`
+* `.spec.provider.workers[].volume.size`
+* `.spec.provider.workers[].providerConfig`
+* `.spec.provider.workers[].cri`
```

Comment:
Maybe it would be better for the user to have all rolling update triggers in one place instead. 
Maybe we can add the OS version and K8s version here too. WDYT